server/backend/api/app/crud/validator/user_validator.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import copy
+
+from api.app.integration.idim_proxy import IdimProxyService
+from api.app.constants import (
+    IdimSearchUserParamType,
+    UserType,
+    ERROR_CODE_INVALID_REQUEST_PARAMETER,
+)
+from api.app.schemas import (
+    IdimProxySearchParam,
+    IdimProxyBceidSearchParam,
+    Requester,
+    TargetUser,
+)
+from api.app.utils import utils
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class UserValidator:
+    def __init__(self, requester: Requester, target_user: TargetUser):
+        self.verified_target_user = copy.deepcopy(target_user)
+        self.idim_proxy_service = IdimProxyService(requester)
+
+    def verify_user_exist(self) -> TargetUser:
+        search_result = None
+        if self.verified_target_user.user_type_code == UserType.IDIR:
+            # IDIM web service doesn't support search IDIR by user_guid, so we search by userID
+            search_result = self.idim_proxy_service.search_idir(
+                IdimProxySearchParam(**{"userId": self.verified_target_user.user_name})
+            )
+
+            # in edge case, the return guid from search doesn't match the guid given from request parameter
+            # this is unlikely to happen if the request comes from frontend because we also validate user in frontend
+            # but could happen if make backend api call directly
+            if (
+                search_result.get("found")
+                and search_result.get("guid").lower()
+                != self.verified_target_user.user_guid.lower()
+            ):
+                error_msg = (
+                    f"Invalid request, found user {self.verified_target_user.user_name} with user type {self.verified_target_user.user_type_code}, "
+                    f"but found user guid {search_result.get('guid').lower()} does not match the user guid in request {self.verified_target_user.user_guid.lower()}"
+                )
+                utils.raise_http_exception(
+                    error_code=ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                    error_msg=error_msg,
+                )
+
+        elif self.verified_target_user.user_type_code == UserType.BCEID:
+            # search_result = self.idim_proxy_service.search_business_bceid(
+            #     IdimProxyBceidSearchParam(
+            #         **{
+            #             "searchUserBy": IdimSearchUserParamType.USER_GUID,
+            #             "searchValue": self.verified_target_user.user_guid,
+            #         }
+            #     )
+            # )
+
+            # currently bceid cannot search bceid by user guid in TEST env, we just search by user name for now
+            search_result = self.idim_proxy_service.search_business_bceid(
```

Comment:
Currently the bceid user cannot verify bceid user by user_guid, even in test env, use username for now. Will update to use user_guid after we enabled it.

Reply:
actually this is enabled in PROD env, we just need it to be enabled in test env as well

Reply:
Thank you Catherine for the information. 
Since this is commented out, should we add comment in ticket 1181 at your TODO to remind that also this part of code need to be changed after test env is enabled as a follow-up change?

Reply:
Hi Ian, I already added the comment in 1181 for all the scenarios that are not working and I already sent the question to IDIM.